g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -180,5 +180,36 @@ public APIResponseDTO checkTranslationResult(String productName, String componen
 						en);
 		}
 		return m;
-	}	
+	}
+
+	/**
+	 * Get the the matched version from the version list by comparing to the input version
+	 *
+	 * @param productName
+	 * @param version
+	 * @return a matched version, if there's no matched version then return input version
+	 */
+	private String getMatchedVersion(String productName, String version) throws L3APIException{
+		int targetVersion = Integer.parseInt(version.replace(".", ""));
+		Map<String, String[]> productsAndVersions = productService.getProductsAndVersions();
+		int matchedIntVersion = 0;
+		String matchedVersion = "";
+		if(productsAndVersions != null && !productsAndVersions.isEmpty()) {
+			String[] versionList = productsAndVersions.get(productName);
+			if(versionList != null && versionList.length > 0) {
+				for(String s : versionList) {
+					int sourceVersion = Integer.parseInt(s.replace(".", ""));
```

Comment:
What if version name contains characters, such as 7.1beta?
What if souceVersion are 3.4, 3.35, but target version is 3.5? Which one wil be the mached version?

Reply:
We normalize the version naming, so no 7.1beat situation.
The second comment had been fixed.